webapp/app/routes/app/emergency/$id/_route.tsx

```diff
@@ -0,0 +1,239 @@
+              {/*  <span>Proposed Version:</span>*/}
+              {/*  <span className="w-1/2 break-words text-right">new version</span>*/}
+              {/*</div>*/}
+              <div className="flex justify-between">
+                <span>Title:</span>
+                <span className="w-4/5 justify-end break-words text-right">{proposal.title}</span>
+              </div>
+              <div className="flex justify-between">
+                <span>Proposal ID:</span>
+                <span className="w-4/5 justify-end break-words text-right">
+                  {proposal.externalId}
+                </span>
+              </div>
+              <div className="flex justify-between">
+                <span>Proposed On:</span>
+                <div className="flex w-3/4 flex-col break-words text-right">
+                  <span>{new Date(proposal.proposedOn).toLocaleDateString()}</span>
+                  <span>{new Date(proposal.proposedOn).toLocaleTimeString()}</span>
+                </div>
+              </div>
+            </div>
+          </CardContent>
+        </Card>
+        <Card className="pb-10">
+          <CardHeader className="pt-7">
+            <CardTitle>Proposal Status</CardTitle>
+          </CardHeader>
+          <CardContent>
+            <div className="space-y-5">
+              <StatusIndicator
+                label="Security Council Approvals"
+                signatures={gatheredScSignatures}
+                necessarySignatures={SC_THRESHOLD}
+              />
+              <StatusIndicator
+                label="Guardian Approvals"
+                signatures={gatheredGuardianSignatures}
+                necessarySignatures={GUARDIAN_THRESHOLD}
+              />
+              <StatusIndicator
+                label="ZkFoundation approvals"
+                signatures={gatheredZkFoundationSignatures}
+                necessarySignatures={ZK_FOUNDATION_THRESHOLD}
+              />
+            </div>
+          </CardContent>
+        </Card>
+        <Card className="pb-10">
+          <CardHeader>
+            <CardTitle>Signatures</CardTitle>
+          </CardHeader>
+          <CardContent className="flex flex-col space-y-3">
+            <SignButton
+              proposalId={proposal.externalId}
+              contractData={{
+                actionName: actionName,
+                address: addresses.emergencyBoard,
+                name: "EmergencyUpgradeBoard",
+              }}
+              disabled={haveAlreadySigned}
+              postAction={$path("/app/emergency/:id", { id: proposal.externalId })}
+            >
+              Approve
+            </SignButton>
+          </CardContent>
+        </Card>
+        <Card className="pb-10">
+          <CardHeader>
+            <CardTitle>Broadcast actions</CardTitle>
+          </CardHeader>
+          <CardContent className="flex flex-col space-y-3">
+            <ExecuteEmergencyUpgradeButton
```

Comment:
Rather than client side logic whether to determine whether to enable button or not, what do you reckon about just reading the emergencyProposal status column if READY or not? That way DB is source of truth in a single place